Support castanets for android port.

suyambulingamrm opened the pull request:
This CL helps to run castanets in android port. It also handles
1. Difference in WebPreference Mojo IPC.
2. Few changes that enable cross platform communication.

Ways to launch renderer on android target.
Type 1:
./out/myandroid/bin/chrome_public_apk  launch --args='--type=renderer --server-address=IP'
Type 2:
In adb shell => echo "_ --type=renderer --server-address=\"IP\"" > /data/local/tmp/chrome-command-line
adb shell am start -a android.intent.action.VIEW -n org.chromium.chrome/org.chromium.chrome.browser.firstrun.FirstRunActivity --es args '"--type=renderer"'

Signed-off-by: suyambu.rm <suyambu.rm@samsung.com>

suyambulingamrm commented:
Latest patch set removes the dependency of hardcoding css style for android theme.
The issue was, pak files was accessed using file descriptor in android renderer which is shared by browser process. This wont work in our case.
So we again reopen resources.pak in castanets mode.
https://github.com/Samsung/Castanets/pull/70/commits/b196bbf465d95974db8e29a3ba74efacef472139#diff-865b406493b56792788a9accfec60e35

hyowon commented:
This patch seems to contain all changes related to several issues.
Please do not combine commits before pr. It makes review difficult.
If necessary, we can squash them with a single commit after the review and before merge.

Could you split this into multiple commits for each issue?

Commit:
Make autofill driver mojo similar to tizen environment.

Autofill driver mojo in tizen environment has additional setting.
So it is necessary to make castanets and tizen environment similar.

Signed-off-by: suyambu.rm <suyambu.rm@samsung.com>

Commit:
Make CreateViewParams in renderer mojo similar to tizen
environment.

CreateViewParams mojo message processing will fail, if the params
are not similar as tizen environment.

Signed-off-by: suyambu.rm <suyambu.rm@samsung.com>

Commit:
Make Gpu config similar to tizen environment.

The changes are necessary to make gpu channel communication
between castanets and tizen environment.

Signed-off-by: suyambu.rm <suyambu.rm@samsung.com>

Commit:
Do not use ChromeViewHostMsg_AllowDOMStorage IPC.

ChromeViewHostMsg_AllowDOMStorage is an unhandled IPC with respect to tizen.
So dropping the same.

Signed-off-by: suyambu.rm <suyambu.rm@samsung.com>

Commit:
Support castanets for android port.

This change supports castanets for android environment.
Ways to launch renderer on android target.
Type 1:
./out/myandroid/bin/chrome_public_apk launch --args='--type=renderer --server-address=IP'
Type 2:
In adb shell => echo "_ --type=renderer --server-address="IP"" > /data/local/tmp/chrome-command-line
adb shell am start -a android.intent.action.VIEW -n org.chromium.chrome/org.chromium.chrome.browser.firstrun.FirstRunActivity --es args '"--type=renderer"'

Signed-off-by: suyambu.rm suyambu.rm@samsung.com

Commit:
Make WebPreference IPC params same for all target profiles.

This change is necessary for a successful read of WebPreference IPC
across all platforms.

Signed-off-by: suyambu.rm <suyambu.rm@samsung.com>

suyambulingamrm commented:
> This patch seems to contain all changes related to several issues.
> Please do not combine commits before pr. It makes review difficult.
> If necessary, we can squash them with a single commit after the review and before merge.
> 
> Could you split this into multiple commits for each issue?

Yes, I agree. Added multiple commits in this PR.

is46-kim commented on skia/config/SkUserConfig.h:
```diff
@@ -141,7 +141,12 @@ SK_API void SkDebugf_FileLine(const char* file, int line, bool fatal,
     "%s:%d: failed assertion \"%s\"\n", \
     __FILE__, __LINE__, #cond); } } while (false)
 
-#if !defined(ANDROID)   // On Android, we use the skia default settings.
```
How about adding only condition instead of duplicate code?
#if !defined(ANDROID) || defined(CASTANETS)

Commit:
Make TextInputState struct same to tizen environment.

TextInputState is used when ime is opened. This change is needed
for ime to show up.

Signed-off-by: suyambu.rm <suyambu.rm@samsung.com>

suyambulingamrm commented:
Added a new patch for ime to show up.
https://github.com/Samsung/Castanets/pull/70/commits/50f3812475f2c625c0e548c77223007672f580d3

Commit:
Send InputHostMsg_DidHandleKeyEvent from renderer.

InputHostMsg_DidHandleKeyEvent IPC has to be sent to browser process
in case of tizen environment to handle key event.

Signed-off-by: suyambu.rm <suyambu.rm@samsung.com>

younghajung reviewed (commented):
This commit looks good to me.

Commit:
Fix dispatch error for GpuCommandBufferMsg_CreateImage IPC between
tizen and android renderer.

GpuCommandBufferMsg_CreateImage has difference in its param between
android and tizen environment. It results in IPC dispatch error
and "Image with ID doesn't exist" log is thrown from CommandBufferStub.

Signed-off-by: suyambu.rm <suyambu.rm@samsung.com>

suyambulingamrm commented:
https://github.com/Samsung/Castanets/pull/70/commits/b386eb6ce03d285d401a6789434acc9873b1f9ec
=> For handling key events when running renderer on android.
https://github.com/Samsung/Castanets/pull/70/commits/db531f8674f6a3a4b4e9c4a58533355595bc1441
=> I/CHROMIUM( 3703): [INFO:ipc_message_templates.h(149)] IPC Dispatch error for GpuCommandBufferMsg_CreateImage TYPE: 307285317 ID: 307285317
E/CHROMIUM( 3703): [ERROR:gpu_command_buffer_stub.cc(1249)] Image with ID doesn't exist.

younghajung reviewed (commented):
5a75c97 is looks good to me.

hyowon commented on gpu/command_buffer/common/capabilities.h:
```diff
@@ -152,6 +152,8 @@ struct GPU_EXPORT Capabilities {
   bool flips_vertically = false;
   bool msaa_is_slow = false;
   bool disable_one_component_textures = false;
+  bool disable_multisampling_color_mask_usage = false;
```
#if defined(CASTANETS) ?

djmixkim commented on content/renderer/input/render_widget_input_handler.cc:
```diff
@@ -265,6 +269,21 @@ void RenderWidgetInputHandler::HandleInputEvent(
       processed = widget_->GetWebWidget()->HandleInputEvent(coalesced_event);
   }
 
+#if defined(CASTANETS)
```
This changes will make duplicated behavior with line 233 for android codes whenever calling WebInputEvent::IsKeyboardEventType().

Is it OK?

suyambulingamrm replied:
Yes you are right. It may cause some conflict. The original condition specifically handles DPAD_CENTER key and text input. Not sure about its background.
I just added a TODO comment regarding the specific handling of DPAD_CENTER key.

djmixkim commented:
50f3812475f2c625c0e548c77223007672f580d3 is looks good to me

is46-kim commented on ui/gfx/ipc/gfx_param_traits_macros.h:
```diff
@@ -39,18 +39,23 @@ IPC_STRUCT_TRAITS_BEGIN(gfx::GpuMemoryBufferHandle)
+  IPC_STRUCT_TRAITS_MEMBER(tbm_surface)
+  IPC_STRUCT_TRAITS_MEMBER(media_packet)
+#endif
+
```
remove a blank line

is46-kim commented:
The below commits look good to me.
c6411bf, 461c2d4, a30ac81, 727b9fd, 50f3812

Sunwoo-Nam commented:
Please check conflicts.

suyambulingamrm commented:
> Please check conflicts.

Sure. Done.
